python/nav/smidumps/JUNIPER-ALARM-MIB.py

```diff
@@ -0,0 +1,239 @@
+                "type" : { "module" :"SNMPv2-SMI", "name" : "Gauge32"},
+            },
+            "access" : "readonly",
+            "description" :
+                """The number of currently active and non-silent 
+yellow alarms.
+
+This object is independent of the ACO/LT (Alarm
+Cut Off / Lamp Test) button.""",
+        }, # scalar
+        "jnxYellowAlarmLastChange" : {
+            "nodetype" : "scalar",
+            "moduleName" : "JUNIPER-ALARM-MIB",
+            "oid" : "1.3.6.1.4.1.2636.3.4.2.2.3",
+            "status" : "current",
+            "syntax" : {
+                "type" : { "module" :"SNMPv2-TC", "name" : "TimeStamp"},
+            },
+            "access" : "readonly",
+            "description" :
+                """The value of sysUpTime when the yellow alarm
+last changed - either from off to on or vice
+versa. 	Zero if unknown or never changed since
+the agent was up.""",
+        }, # scalar
+        "jnxRedAlarms" : {
+            "nodetype" : "node",
+            "moduleName" : "JUNIPER-ALARM-MIB",
+            "oid" : "1.3.6.1.4.1.2636.3.4.2.3",
+        }, # node
+        "jnxRedAlarmState" : {
+            "nodetype" : "scalar",
+            "moduleName" : "JUNIPER-ALARM-MIB",
+            "oid" : "1.3.6.1.4.1.2636.3.4.2.3.1",
+            "status" : "current",
+            "syntax" : {
+                "type" :                 {
+                    "basetype" : "Enumeration",
+                    "other" : {
+                        "nodetype" : "namednumber",
+                        "number" : "1"
+                    },
+                    "off" : {
+                        "nodetype" : "namednumber",
+                        "number" : "2"
+                    },
+                    "on" : {
+                        "nodetype" : "namednumber",
+                        "number" : "3"
+                    },
+                },
+            },
+            "access" : "readonly",
+            "description" :
+                """The red alarm indication on the craft interface 
+panel.
+
+The red alarm is on when there is some system 
+failure or power supply failure or the system 
+is experiencing a hardware malfunction or some 
+threshold is being exceeded.
+
+This red alarm state could be turned off by the
+ACO/LT (Alarm Cut Off / Lamp Test) button on the
+front panel module.""",
+        }, # scalar
+        "jnxRedAlarmCount" : {
+            "nodetype" : "scalar",
+            "moduleName" : "JUNIPER-ALARM-MIB",
+            "oid" : "1.3.6.1.4.1.2636.3.4.2.3.2",
+            "status" : "current",
+            "syntax" : {
+                "type" : { "module" :"SNMPv2-SMI", "name" : "Gauge32"},
+            },
+            "access" : "readonly",
+            "description" :
+                """The number of currently active and non-silent 
+red alarms.
+
+This object is independent of the ACO/LT (Alarm
+Cut Off / Lamp Test) button.""",
+        }, # scalar
+        "jnxRedAlarmLastChange" : {
```

Comment:
If it does turn out that the non-stateful behavior of this plugin causes logging storms in eventengine, it might be a good idea to collect `jnxYellowAlarmLastChange` and `jnxRedAlarmLastChange` and persist these, so that we avoid any collection/event posting if these timestamps are unchanged.

This is a common use case, and if an implementation is needed, please refer to how `TimestampChecker` is being used in other plugins:

https://github.com/Uninett/nav/blob/e6634e512c8ecf283c85a701366620e724806ab7/python/nav/ipdevpoll/timestamps.py#L30-L55

Reply:
As just discussed in person: This idea is scrapped since `jnxRedAlarmLastChange` and `jnxYellowAlarmLastChange` only indicate if the alarm state was changed from on to off or off to on, but not if the count changed from 1 to 2 for example. 

We will instead collect the alarm count every time, but save it in `NetboxInfo` and only post an event if this value changed.